optimum/intel/openvino/quantization.py

```diff
@@ -384,33 +383,38 @@ def data_collator(batch):
         else:
             onnx_config = onnx_config_class(model.config)
 
-        onnx_path = save_directory / ONNX_WEIGHTS_NAME
-
-        # Export the model to the ONNX format
-        opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
-        opset = max(opset, MIN_ONNX_QDQ_OPSET)
-        export(
-            model=compressed_model,
-            config=onnx_config,
-            opset=opset,
-            output=onnx_path,
-        )
+        model_path = save_directory / (ONNX_WEIGHTS_NAME if quantization_config.save_onnx_model else OV_XML_FILE_NAME)
+        if quantization_config.save_onnx_model:
+            # Export the model to the ONNX format
+            opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
+            opset = max(opset, MIN_ONNX_QDQ_OPSET)
+            onnx_export(
+                model=compressed_model,
+                config=onnx_config,
+                opset=opset,
+                output=model_path,
+            )
 
-        # Load and save the compressed model
-        model = core.read_model(onnx_path)
-        self._save_pretrained(model, output_path)
+            # Load and save the compressed model
+            model = core.read_model(model_path)
+            self._save_pretrained(model, output_path)
+        else:
+            _, _, is_onnx = export(model=compressed_model, config=onnx_config, output=output_path)
+            if is_onnx:
+                onnx_path = save_directory / ONNX_WEIGHTS_NAME
```

Comment:
It seems like the logic within this "if" is not required except `os.remove()` stuff.

Reply:
reworked